src/internal/Notification.ts

```diff
@@ -181,16 +154,16 @@ export class Notification<T> {
    * being removed as it has limited usefulness, and we're trying to streamline the library.
    */
   toObservable(): Observable<T> {
-    const kind = this.kind;
-    switch (kind) {
-      case 'N':
-        return of(this.value!);
-      case 'E':
-        return throwError(this.error);
-      case 'C':
-        return EMPTY;
-    }
-    throw new Error('unexpected notification kind value');
+    const { kind, value, error } = this;
+    return kind === 'N'
+      ? of(value!)
+      : kind === 'E'
+      ? throwError(error)
+      : kind === 'C'
+      ? EMPTY
+      : (() => {
+          throw new TypeError(`Unexpected notification kind ${kind}`);
+        })();
```

Comment:
IMO this is taking things too far sacrificing readability for a few bytes, especially after gzip.

Reply:
Probably. I've refactored it a little and added comments to help.